apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -491,7 +491,18 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
             if (SendJsonPayload)
             {
                 var jsonStruct = new Struct();
-                jsonStruct.Fields.Add("message", Value.ForString(RenderLogEvent(Layout, loggingEvent)));
+                if (UseNLogCustomJsonLayout)
```

Comment:
Instead of "just" replacing the "message", then I think it should replace the entire JsonPayload.

So the output from the JsonLayout becomes the JsonPayload. This gives total freedom, which you are chasing.

Reply:
Nah I prefer that it is all or nothing when deciding to use JsonLayout for customizing JsonPayload. In one month another person will show up and say that "properties"-field should placed in a completely different way. Now this person will look at your code, and recognize that even more strange options has to implemented to control with it should be `default` - `special mix`- `new special mix` - etc.

Just because some random Java-library have decide to lock itself into its own corner, then it doesn't mean all others have to do the same. But no-matter it is not my decision to make, since I'm not the owner of this library.

Reply:
Not a random library, but google one. I actually also don't have any idea what is the approach we should follow. For me, there is no problem to add service context properties by myself, but should we ignore context properties totally? 
But I get your point of view, it's reasonable.